facturacion_mexico/facturacion_fiscal/tests/test_cancel_guard_draft_ffm.py

```
@@ -0,0 +1,217 @@
+
+		# El Response Log de la draft permanece
+		self.assertTrue(frappe.db.exists("FacturAPI Response Log", self.response_log))
+
+		# La FFM fiscal válida SÍ se desvinculó (para permitir el cancel nativo de la SI)
+		self.assertEqual(frappe.db.get_value("Factura Fiscal Mexico", self.ffm_valida, "sales_invoice"), "")
+
+	def test_draft_con_uuid_sigue_bloqueando(self):
+		# Convertir la draft en una con UUID → vuelve a ser fiscalmente activa y BLOQUEA
+		frappe.db.set_value("Factura Fiscal Mexico", self.ffm_draft, "fm_uuid", "U-CONUUID")
+		with mock.patch("frappe.model.document.Document.cancel"):
+			with self.assertRaises(frappe.ValidationError):
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)